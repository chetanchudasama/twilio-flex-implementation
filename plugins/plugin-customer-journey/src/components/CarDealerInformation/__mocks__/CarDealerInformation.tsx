import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
