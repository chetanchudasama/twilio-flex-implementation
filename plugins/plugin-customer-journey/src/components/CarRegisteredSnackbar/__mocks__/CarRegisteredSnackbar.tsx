import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
