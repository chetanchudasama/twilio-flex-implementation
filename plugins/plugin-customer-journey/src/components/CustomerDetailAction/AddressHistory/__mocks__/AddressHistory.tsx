import React from "react";
import { AddressHistoryProps } from "../AddressHistory";

const AddressHistory: React.FC<AddressHistoryProps> = (props) => (
  <div className="address-history" />
);

export default AddressHistory;
