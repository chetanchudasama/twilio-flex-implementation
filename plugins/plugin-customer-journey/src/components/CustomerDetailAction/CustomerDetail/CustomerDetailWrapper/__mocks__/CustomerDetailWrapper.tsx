import React from "react";
import { ComponentProps } from "../CustomerDetailWrapper";

const CustomerDetailWrapper: React.FC<ComponentProps> = (props) => (
  <div className="customer-detail-wrapper" />
);

export default CustomerDetailWrapper;
