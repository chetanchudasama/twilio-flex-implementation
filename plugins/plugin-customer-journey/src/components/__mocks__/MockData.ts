import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
