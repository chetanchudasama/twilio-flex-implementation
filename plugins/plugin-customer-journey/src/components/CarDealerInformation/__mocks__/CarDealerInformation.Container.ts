import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
