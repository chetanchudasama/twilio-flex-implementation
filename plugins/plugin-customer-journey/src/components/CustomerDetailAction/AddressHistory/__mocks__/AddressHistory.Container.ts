import AddressHistory from "./AddressHistory";

export default AddressHistory;
