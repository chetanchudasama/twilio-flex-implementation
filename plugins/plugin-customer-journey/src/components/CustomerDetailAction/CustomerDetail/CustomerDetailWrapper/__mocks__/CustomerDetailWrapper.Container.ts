import CustomerDetailWrapper from "./CustomerDetailWrapper";

export default CustomerDetailWrapper;
