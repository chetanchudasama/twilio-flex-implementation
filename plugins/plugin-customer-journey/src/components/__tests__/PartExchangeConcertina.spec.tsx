import { Button } from "@material-ui/core";
import { Manager } from "@twilio/flex-ui";
import { shallow, mount } from "enzyme";
import flushPromises from "flush-promises";
import React from "react";
import { act } from "react-dom/test-utils";
import { create } from "react-test-renderer";
import {
  OwnProps,
  PartExchangeConcertina,
} from "../PartExchangeConcertina/PartExchangeConcertina";
import { flexManagerMock } from "../__mocks__/MockData";

const props: OwnProps = {
  partExchangeRegistration: "REG1 234",
};

const mockGetPXResponse = jest.fn(() =>
  Promise.resolve({
    tradeValue: 1,
    retailTransacted: 2,
    retailValue: 3,
  })
);

Manager.getInstance = flexManagerMock;

jest.mock("../../services/application.service", () => ({
  useApplicationService: jest.fn(() => ({
    getPartExchangeValuation: mockGetPXResponse,
  })),
}));

describe("PartExchangeConcertina", () => {
  it("renders correctly", async () => {
    expect(
      create(<PartExchangeConcertina {...props} />).toJSON()
    ).toMatchSnapshot();
  });

  it("should have -- before button is clicked / api is called", () => {
    const wrapper = shallow(<PartExchangeConcertina {...props} />);
    const transacted = wrapper.find({ id: "px-retail-transacted" }).text();
    const glassTrade = wrapper.find({ id: "px-glass-trade" }).text();
    const askingValue = wrapper.find({ id: "px-retail-asking" }).text();

    expect(
      [transacted, glassTrade, askingValue].every((e) => e === "--")
    ).toBeTruthy();
  });

  it("should call api when the Value PX button is clicked", async () => {
    const wrapper = mount(<PartExchangeConcertina {...props} />);

    const input = wrapper.find(".mileage-container").find("input");
    input.simulate("change", { target: { value: "100" } });
    wrapper.update();

    await act(async () => {
      wrapper.find(Button).at(0).simulate("click");
      wrapper.update();
      await flushPromises();
    });
    expect(mockGetPXResponse).toHaveBeenCalledTimes(1);
  });

  it("should display information when api is called", async () => {
    const wrapper = mount(<PartExchangeConcertina {...props} />);

    const input = wrapper.find(".mileage-container").find("input");
    input.simulate("change", { target: { value: "100" } });
    wrapper.update();

    await act(async () => {
      wrapper.find(Button).at(0).simulate("click");
      await flushPromises();
      expect(mockGetPXResponse).toHaveBeenCalled();
    });
    wrapper.update();

    const glassTrade = wrapper.find({ id: "px-glass-trade" });
    expect(glassTrade.exists()).toBeTruthy();
    expect(glassTrade.text()).toEqual("£1");

    const transacted = wrapper.find({ id: "px-retail-transacted" });
    expect(transacted.exists()).toBeTruthy();
    expect(transacted.text()).toEqual("£2");

    const askingValue = wrapper.find({ id: "px-retail-asking" });
    expect(askingValue.exists()).toBeTruthy();
    expect(askingValue.text()).toEqual("£3");
  });
});
