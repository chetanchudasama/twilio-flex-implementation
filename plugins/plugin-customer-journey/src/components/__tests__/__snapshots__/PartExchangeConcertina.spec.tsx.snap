// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartExchangeConcertina renders correctly 1`] = `
<div
  className="css-171c6wf"
>
  <div
    className="MuiPaper-root-5 MuiPaper-elevation1-8 MuiPaper-rounded-6 MuiExpansionPanel-root-1 MuiExpansionPanel-rounded-2"
  >
    <div
      aria-expanded={false}
      className="MuiButtonBase-root-38 MuiExpansionPanelSummary-root-32"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiExpansionPanelSummary-content-36"
      >
        <div>
          Value PX
        </div>
      </div>
      <div
        aria-hidden="true"
        className="MuiButtonBase-root-38 MuiIconButton-root-41 MuiExpansionPanelSummary-expandIcon-37"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label-46"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-47 MuiSvgIcon-colorSecondary-49"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-242"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      className="MuiCollapse-container-56"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper-58"
      >
        <div
          className="MuiCollapse-wrapperInner-59"
        >
          <div
            className="MuiExpansionPanelDetails-root-60"
          >
            <div
              className="MuiGrid-container-61 MuiGrid-item-62 MuiGrid-grid-xs-12-101"
            >
              <div
                className="MuiGrid-item-62 MuiGrid-grid-xs-5-94 vrm-container"
              >
                <div
                  className="MuiFormControl-root-162 MuiFormControl-marginDense-164 MuiFormControl-fullWidth-165 CustomInputContent-formControl-158"
                >
                  <label
                    className="MuiFormLabel-root-177 MuiFormLabel-filled-181 MuiInputLabel-root-166 MuiInputLabel-formControl-171 MuiInputLabel-animated-174 MuiInputLabel-shrink-173 MuiInputLabel-marginDense-172 CustomInputContent-label-159"
                    data-shrink={true}
                  >
                    VRM
                  </label>
                  <div
                    className="MuiInputBase-root-197 MuiOutlinedInput-root-184 MuiInputBase-formControl-198 MuiInputBase-marginDense-204 CustomInputContent-input-160 "
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="MuiPrivateNotchedOutline-root-214 MuiOutlinedInput-notchedOutline-191"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="MuiPrivateNotchedOutline-legend-215"
                        style={
                          Object {
                            "width": 0,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input-207 MuiOutlinedInput-input-192 MuiInputBase-inputMarginDense-208 MuiOutlinedInput-inputMarginDense-193"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value="REG1 234"
                    />
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-item-62 MuiGrid-grid-xs-5-94 mileage-container"
              >
                <div
                  className="MuiFormControl-root-162 MuiFormControl-marginDense-164 MuiFormControl-fullWidth-165 CustomInputContent-formControl-158"
                >
                  <label
                    className="MuiFormLabel-root-177 MuiInputLabel-root-166 MuiInputLabel-formControl-171 MuiInputLabel-animated-174 MuiInputLabel-shrink-173 MuiInputLabel-marginDense-172 CustomInputContent-label-159"
                    data-shrink={true}
                  >
                    Mileage
                  </label>
                  <div
                    className="MuiInputBase-root-197 MuiOutlinedInput-root-184 MuiInputBase-formControl-198 MuiInputBase-marginDense-204 CustomInputContent-input-160 "
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="MuiPrivateNotchedOutline-root-214 MuiOutlinedInput-notchedOutline-191"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="MuiPrivateNotchedOutline-legend-215"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input-207 MuiOutlinedInput-input-192 MuiInputBase-inputMarginDense-208 MuiOutlinedInput-inputMarginDense-193"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-item-62 MuiGrid-grid-xs-2-91 value-px-container"
              >
                <button
                  className="MuiButtonBase-root-38 MuiButtonBase-disabled-39 MuiButton-root-216 MuiButton-text-218 MuiButton-flat-221 MuiButton-disabled-236 value-px-btn"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="-1"
                  type="button"
                >
                  <span
                    className="MuiButton-label-217"
                  >
                    Value PX
                  </span>
                </button>
              </div>
              <div
                className="MuiGrid-container-61 MuiGrid-item-62 MuiGrid-grid-xs-12-101 panel-second-row"
              >
                <div
                  className="MuiGrid-item-62 MuiGrid-grid-xs-4-93"
                >
                  <label
                    className="MuiFormLabel-root-177 MuiInputLabel-root-166 MuiInputLabel-animated-174"
                  >
                    Glass's Trade
                  </label>
                  <div
                    className="value-px-item"
                    id="px-glass-trade"
                  >
                    --
                  </div>
                </div>
                <div
                  className="MuiGrid-item-62 MuiGrid-grid-xs-4-93"
                >
                  <label
                    className="MuiFormLabel-root-177 MuiInputLabel-root-166 MuiInputLabel-animated-174"
                  >
                    Retail Transacted
                  </label>
                  <div
                    className="value-px-item"
                    id="px-retail-transacted"
                  >
                    --
                  </div>
                </div>
                <div
                  className="MuiGrid-item-62 MuiGrid-grid-xs-4-93"
                >
                  <label
                    className="MuiFormLabel-root-177 MuiInputLabel-root-166 MuiInputLabel-animated-174"
                  >
                    Retail Asking
                  </label>
                  <div
                    className="value-px-item"
                    id="px-retail-asking"
                  >
                    --
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
